diff --git a/configs/bootanimation.mk b/configs/bootanimation.mk
index 1abc514..2dc5ae6 100644
--- a/configs/bootanimation.mk
+++ b/configs/bootanimation.mk
@@ -37,7 +37,7 @@ ifneq ($(filter pa_amami pa_nozomi pa_falcon pa_peregrine pa_ghost pa_lettuce pa
 endif
 
 # PA HDPI Devices
-ifneq ($(filter pa_condor pa_p3100 pa_p3110 pa_taoshan,$(TARGET_PRODUCT)),)
+ifneq ($(filter pa_janice pa_condor pa_p3100 pa_p3110 pa_taoshan,$(TARGET_PRODUCT)),)
     PRODUCT_COPY_FILES += \
         vendor/pa/prebuilt/bootanimation/800x480.zip:system/media/bootanimation.zip
 endif
diff --git a/products/AndroidProducts.mk b/products/AndroidProducts.mk
index 4e19ea4..844f17e 100644
--- a/products/AndroidProducts.mk
+++ b/products/AndroidProducts.mk
@@ -198,3 +198,6 @@ endif
 ifeq (pa_Z008,$(TARGET_PRODUCT))
     PRODUCT_MAKEFILES += $(LOCAL_DIR)/pa_Z008.mk
 endif
+ifeq (pa_janice,$(TARGET_PRODUCT))
+    PRODUCT_MAKEFILES += $(LOCAL_DIR)/pa_janice.mk
+endif
diff --git a/products/pa_janice.mk b/products/pa_janice.mk
new file mode 100644
index 0000000..7c3b73a
--- /dev/null
+++ b/products/pa_janice.mk
@@ -0,0 +1,44 @@
+# Copyright (C) 2015 ParanoidAndroid Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Check for target product
+
+ifeq (pa_janice,$(TARGET_PRODUCT))
+
+# OVERLAY_TARGET adds overlay asset source
+OVERLAY_TARGET := pa_hdpi
+
+# Build paprefs from sources
+PREFS_FROM_SOURCE ?= false
+
+# Inherit telephony common stuff
+$(call inherit-product, vendor/pa/configs/telephony.mk)
+
+# Include AOSPA common configuration
+include vendor/pa/main.mk
+
+# Inherit device configuration
+$(call inherit-product, device/samsung/janice/full_janice.mk)
+
+# Override AOSP build properties
+PRODUCT_DEVICE := janice
+PRODUCT_NAME := pa_janice
+PRODUCT_BRAND := samsung
+PRODUCT_MODEL := GT-I9070
+PRODUCT_MANUFACTURER := samsung
+
+# Set build fingerprint / ID / Product Name ect.
+PRODUCT_BUILD_PROP_OVERRIDES += PRODUCT_NAME=GT-I9070 TARGET_DEVICE=janice
+
+endif
diff --git a/vendorsetup.sh b/vendorsetup.sh
index 370c927..c1bc35b 100644
--- a/vendorsetup.sh
+++ b/vendorsetup.sh
@@ -81,3 +81,4 @@ add_lunch_combo pa_z3-userdebug
 add_lunch_combo pa_z3c-userdebug
 add_lunch_combo pa_Z00A-userdebug
 add_lunch_combo pa_Z008-userdebug
+add_lunch_combo pa_janice-userdebug
-- 
2.1.0

